src/main/java/de/tum/in/www1/artemis/service/ZipFileService.java

```diff
@@ -0,0 +1,81 @@
+package de.tum.in.www1.artemis.service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ZipFileService {
+
+    private final Logger log = LoggerFactory.getLogger(ZipFileService.class);
+
+    /**
+     * Create a zip file of the given paths and save it in the zipFilePath
+     *
+     * @param zipFilePath path where the zip file should be saved
+     * @param paths multiple paths that should be zipped
+     * @throws IOException if an error occurred while zipping
+     */
+    public void createZipFile(Path zipFilePath, List<Path> paths) throws IOException {
```

Comment:
Codacy found an issue: [Parameter 'paths' is not assigned and could be declared final](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968852)